plugins/codemirror/plugin.js

```diff
@@ -0,0 +1,130 @@
+(function() {
+	'use strict';
+
+	var stylesLoaded = false;
+
+
+	CKEDITOR.plugins.add('codemirror', {
+		_createCodeMirrorEditor: function(editor) {
+			var instance = this;
+
+			var codemirror = CKEDITOR.plugins.codemirror;
+
+			editor.addMode('source', function(callback) {
+				var contentsSpace = editor.ui.space('contents');
+
+				var textarea = contentsSpace.getDocument().createElement('textarea');
+				contentsSpace.append(textarea);
+
+				instance.codeMirrorEditor = CodeMirror.fromTextArea(
+					textarea.$,
+					{
+						lineNumbers: true,
+						mode: 'text/html'
+					}
+				);
+
+				var oldData = editor.getData(1);
+
+				instance.codeMirrorEditor.setValue(html_beautify(oldData));
+
+				instance.codeMirrorEditor.on('change', instance._handleCodeMirrorChange.bind(instance, editor, oldData));
+
+				editor.on('resize', instance._handleEditorResize.bind(instance));
+
+				editor.fire('ariaWidget', this);
+
+				callback();
+			});
+
+			editor.addCommand('codemirror', codemirror.commands.source);
+		},
+
+		_handleCodeMirrorChange: function(editor, oldData) {
+			var newData = this.codeMirrorEditor.getValue();
+
+			if (newData !== oldData) {
+				editor.setData(newData);
+			}
+		},
+
+		_handleEditorResize: function(event) {
+			this.codeMirrorEditor.setSize(null, event.data.outerHeight);
+		},
+
+		hidpi: true,
+
+		icons: 'source,source-rtl',
+
+		init: function(editor) {
+			var instance = this;
+
+			if (editor.plugins.detectConflict('codemirror', ['sourcearea'])) {
+				return;
+			}
+
+			if (editor.elementMode === CKEDITOR.ELEMENT_MODE_INLINE) {
+				return;
+			}
+
+			if (!stylesLoaded) {
+				CKEDITOR.document.appendStyleSheet(
```

Comment:
Could we use the `deps.json` list here? It will be easy enough to forget to load a dependency since we have to add it both there and here.

Reply:
Not needed now that we're loading a single file :)

Reply:
Nice!!

One could even go as far as to actually directly put the dependencies code on top of the plugin itself in the same file...

![just-sayin](https://user-images.githubusercontent.com/905006/91162894-008ac500-e6cd-11ea-8599-0c0a2a9a420f.gif)

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FfHgKZHDH1BHzi%2Fgiphy.gif&token=stitch" data-gitmeme-token="stitch"><img src="https://media0.giphy.com/media/fHgKZHDH1BHzi/giphy.gif" title="Created by gitme.me with /stitch"/></a>

Reply:
No need to do it, just saying!! 😉

Reply:
![whutchumean](https://user-images.githubusercontent.com/905006/91210393-6f880e00-e70d-11ea-84bc-47810c764e20.gif)

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2F7XIq6GWNwOuUE%2Fgiphy.gif&token=forrest" data-gitmeme-token="forrest"><img src="https://media0.giphy.com/media/7XIq6GWNwOuUE/giphy.gif" title="Created by gitme.me with /forrest"/></a>